yt_dlp/extractor/cnbc.py

```diff
@@ -1,68 +1,114 @@
-                {'force_smil_url': True}),
-            'id': video_id,
-        }
+        display_id = self._match_valid_url(url).group('id')
+        webpage = self._download_webpage(url, display_id)
+        metadata = json.loads(
+            self._html_search_regex(
+                r'window\.__s_data=(\{.*?\});', webpage, 'video JSON data.'))
 
+        link = traverse_obj(
+            metadata,
+            ('page', 'page', 'layout', 1, 'columns', 0, 'modules', 0, 'data', 'encodings', 0, 'url'), expected_type=url_or_none)
 
-class CNBCVideoIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?cnbc\.com(?P<path>/video/(?:[^/]+/)+(?P<id>[^./?#&]+)\.html)'
-    _TEST = {
-        'url': 'https://www.cnbc.com/video/2018/07/19/trump-i-dont-necessarily-agree-with-raising-rates.html',
-        'info_dict': {
-            'id': '7000031301',
-            'ext': 'mp4',
-            'title': "Trump: I don't necessarily agree with raising rates",
-            'description': 'md5:878d8f0b4ebb5bb1dda3514b91b49de3',
-            'timestamp': 1531958400,
-            'upload_date': '20180719',
-            'uploader': 'NBCU-CNBC',
-        },
-        'params': {
-            'skip_download': True,
-        },
-        'skip': 'Dead link',
-    }
+        upload_date = traverse_obj(
+            metadata,
+            ('page', 'page', 'layout', 1, 'columns', 0, 'modules', 0, 'data', 'uploadDate'))
 
-    def _real_extract(self, url):
-        path, display_id = self._match_valid_url(url).groups()
-        video_id = self._download_json(
-            'https://webql-redesign.cnbcfm.com/graphql', display_id, query={
-                'query': '''{
-  page(path: "%s") {
-    vcpsId
-  }
-}''' % path,
-            })['data']['page']['vcpsId']
-        return self.url_result(
-            'http://video.cnbc.com/gallery/?video=%d' % video_id,
-            CNBCIE.ie_key())
+        return {
+            'author': self._html_search_regex(
+                r'<meta name="parsely-author" content="([^<]+)"',
+                webpage,
+                'video author',
+                default=None),
+            'channel': self._html_search_regex(
+                r'<div class="ClipPlayer-clipPlayerIntroSection"><[^>]*>([^<]+)<',
+                webpage,
+                'channel name',
+                fatal=False),
+            'formats': self._extract_akamai_formats(link, display_id),
+            'id': display_id,
+            'upload_date': unified_strdate(upload_date) if upload_date else None,
+            'url': link,
+            **traverse_obj(
+                self._search_json_ld(webpage, display_id, default={}),
+                {
+                    'title': 'title',
+                    'description': 'description',
+                    'duration': 'duration',
+                    'timestamp': 'timestamp',
+                    'thumbnails': 'thumbnails',
+                }),
+        }
```

Comment:
I would simplify this like so and also extract some other values:
```suggestion
        return {
            'id': display_id,
            'display_id': display_id,
            'formats': self._extract_akamai_formats(player_data['playbackURL'], display_id),
            **self._search_json_ld(webpage, display_id, fatal=False),
            **traverse_obj(player_data, {
                'id': ('id', {str_or_none}),
                'title': ('title', {str}),
                'description': ('description', {str}),
                'author': ('author', ..., 'name', {str}),
                'timestamp': ('datePublished', {parse_iso8601}),
                'release_timestamp': ('uploadDate', {parse_iso8601}),
                'modified_timestamp': ('dateLastPublished', {parse_iso8601}),
                'thumbnail': ('thumbnail', {url_or_none}),
                'duration': ('duration', {int_or_none}),
                'channel': ('section', 'title', {str}),
            }, get_all=False),
        }
```
I think it also makes sense to primarily return the actual video id but keep the display_id as a fallback.

Please update the tests acordingly.